File: Ejercicio2.java
public class Ejercicio2 {
    public static void main(String[] args) {
        String salida="";
        int vector[]= new int[20];//vector  de 10 enteros
        for (int i = 0; i < vector.length; i++) {
            vector[i]=(int)(Math.random()*100 + 1);
            System.out.print(vector[i]+"|");
            if (vector[i] % 2  ==0){
                salida+=vector[i]+"| Pos: "+i+"\n";
            }
        }
        System.out.println();
        System.out.println("Los pares son: ");
        System.out.println(salida);
    }
}

File: ejercicio2.py
import random
vector = [];pares = [];posiciones=[]
for i in range(0, 20):
    vector.append(random.randint(1, 100))
    if vector[i] % 2 == 0:
        pares.append(vector[i])
        posiciones.append(i)
print(vector)
print('Los pares son: ',pares)
print('Las posiciones son: ',posiciones)